rpc/src/helpers.rs

```diff
@@ -90,6 +96,89 @@ pub struct BlockHeaderInfo {
     pub content: Option<HeaderContent>,
 }
 
+/// Object containing information to recreate the block header shell information
+#[derive(Serialize, Debug, Clone)]
+pub struct BlockHeaderShellInfo {
+    pub level: i32,
+    pub proto: u8,
+    pub predecessor: String,
+    pub timestamp: String,
+    pub validation_pass: u8,
+    pub operations_hash: String,
+    pub fitness: Vec<String>,
+    pub context: String,
+}
+
+/// Object containing information to recreate the block header shell information
+#[derive(Serialize, Debug, Clone)]
+pub struct BlockHeaderMonitorInfo {
+    pub hash: String,
+    pub level: i32,
+    pub proto: u8,
+    pub predecessor: String,
+    pub timestamp: String,
+    pub validation_pass: u8,
+    pub operations_hash: String,
+    pub fitness: Vec<String>,
+    pub context: String,
+    pub protocol_data: String,
+}
+
+pub struct MonitorHeadStream {
+    pub state: RpcCollectedStateRef,
+    pub last_polled_timestamp: Option<TimeStamp>,
+}
+
+impl Stream for MonitorHeadStream {
+    type Item = Result<String, serde_json::Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Result<String, serde_json::Error>>> {
+        // Note: the stream only ends on the client dropping the connection
+
+        let state = self.state.read().unwrap();
+        let last_update = if let TimeStamp::Integral(timestamp) = state.head_update_time() {
+            *timestamp
+        } else {
+            // TODO: rework this quick hack
```

Comment:
@adonagy 
Ado, what about this?
Will you rework it? If not yet, pls. add here jira reference: `TODO: TE-XYZ rework this quick hack`